File: Exercises/Ex6/returned_answers/RichardFriedrichs/makegauss.py
#!/usr/bin/env python3

import ROOT

def makegauss():
    amount = 1000
    myfile = ROOT.TFile("data2.root", "RECREATE")
    datatree = ROOT.TTree("datatree", "Tree with gaussian data")
    
    # Create a variable to hold the values
    value = ROOT.std.vector('float')()  # Use a vector to hold the value
    datatree.Branch("values", value)
    
    RNG = ROOT.TRandom3(1234)
    
    for i in range(amount):
        value.push_back(RNG.Gaus(15, 3))
        datatree.Fill()

    datatree.Write()

    myfile.Close()

if __name__ == "__main__":
    makegauss()

File: Exercises/Ex6/returned_answers/RichardFriedrichs/makegauss.cpp
#include <TFile.h>
#include <TTree.h>
#include <TH1F.h>
#include <TCanvas.h>
#include <TMath.h>
#include <TRandom3.h>

void makegauss()
{
    Int_t amount = 1000;
    TFile *myfile = TFile::Open("data.root", "RECREATE");
    myfile->cd();
    TTree *datatree = new TTree("datatree", "Tree with gaussian data");
    Float_t value;
    datatree->Branch("values", &value, "values/F");
    TRandom3 RNG(123);
    for (int i = 0; i < amount; ++i)
    {
        value = RNG.Gaus(15,3);
        datatree->Fill();
    }

    datatree->Write();
    myfile->Close();
}

int main(){
    makegauss();
    return 0;
}